packages/lit-node-client-nodejs/src/lib/lit-node-client-nodejs.ts

```diff
@@ -180,115 +174,57 @@ export class LitNodeClientNodeJs
     // usage to a list of delegatee addresses
     const dAppOwnerWalletAddress = ethers.utils.getAddress(
-      await dAppOwnerWallet.getAddress()
+      await params.dAppOwnerWallet.getAddress()
     );
 
-    // -- default configuration for siwe message unless there are arguments
-    const _domain = domain ?? 'example.com';
-    const _expiration =
-      expiration ?? new Date(Date.now() + 1000 * 60 * 7).toISOString();
-    const _statement = '' ?? statement;
-
-    // -- default configuration for recap object capability
-    const _uses = uses ?? '1';
-
     // -- if it's not ready yet, then connect
     if (!this.ready) {
       await this.connect();
     }
 
     // -- validate
-    if (!dAppOwnerWallet) {
+    if (!params.dAppOwnerWallet) {
       throw new Error('dAppOwnerWallet must exist');
     }
 
-    // -- validate dAppOwnerWallet is an ethers wallet
-    // if (!(dAppOwnerWallet instanceof ethers.Wallet || ethers.Signer)) {
-    //   throw new Error('dAppOwnerWallet must be an ethers wallet');
-    // }
-
-    // -- create LitRLIResource
-    // Note: we have other resources such as LitAccessControlConditionResource, LitPKPResource and LitActionResource)
-    // lit-ratelimitincrease://{tokenId}
-    const litResource = new LitRLIResource(capacityTokenId ?? '*');
-
-    const recapObject = await this.generateSessionCapabilityObjectWithWildcards(
-      [litResource]
-    );
-
-    const capabilities = {
-      ...(capacityTokenId ? { nft_id: [capacityTokenId] } : {}), // Conditionally include nft_id
-      ...(delegateeAddresses
-        ? {
-            delegate_to: delegateeAddresses.map((address) =>
-              address.startsWith('0x') ? address.slice(2) : address
-            ),
-          }
-        : {}),
-      uses: _uses.toString(),
-    };
-
-    recapObject.addCapabilityForResource(
-      litResource,
-      LitAbility.RateLimitIncreaseAuth,
-      capabilities
-    );
-
-    // make sure that the resource is added to the recapObject
-    const verified = recapObject.verifyCapabilitiesForResource(
-      litResource,
-      LitAbility.RateLimitIncreaseAuth
-    );
-
-    // -- validate
-    if (!verified) {
-      throw new Error('Failed to verify capabilities for resource');
-    }
-
     const nonce = await this.getLatestBlockhash();
 
-    // -- get auth sig
-    let siweMessage = new siwe.SiweMessage({
-      domain: _domain,
-      address: dAppOwnerWalletAddress,
-      statement: _statement,
-      uri: SIWE_DELEGATION_URI,
-      version: '1',
-      chainId: 1,
-      nonce: nonce?.toString(),
-      expirationTime: _expiration,
+    const siweMessage = await createSiweMessageWithCapacityDelegation({
+      uri: 'lit:capability:delegation',
+      litNodeClient: this,
+      walletAddress: dAppOwnerWalletAddress,
+      nonce: nonce,
+
+      // -- default configuration for recap object capability
+      expiration:
+        params.expiration ?? new Date(Date.now() + 1000 * 60 * 7).toISOString(),
```

Comment:
Can we remove the defaults here and let the defaults defined inside of `createSiweMessage()` do the defaulting?   

It's great that we have a central place defined for creating messages; can we let all defaulting be delegated to that method?  

For example, If we need to default `domain` to `example.com` here instead of letting he `createSiweMessage()` function provide the default, can we add a comment here as to _why_ we want `example.com` to be the default in this codepath, but `localhost` to be the default inside the `createSiweMessage()` method?